### PR TITLE
fix parsing for TCP info on Linux

### PR DESCRIPTION
**Pull request by wfurt**
related to #32727

With this change GetActiveTcpListeners() returns only socket in LISTENING state and GetActiveTcpConnections() shows all but listeners - as documentation states. 

I will add test when I have final fix for #32727.

```
Listeners
0.0.0.0:139
0.0.0.0:80
0.0.0.0:22
127.0.0.1:6010
127.0.0.1:6011
127.0.0.1:6012
127.0.0.1:6013
0.0.0.0:445
127.0.0.1:6014
0.0.0.0:139
0.0.0.0:80
0.0.0.0:22
127.0.0.1:6010
127.0.0.1:6011
127.0.0.1:6012
127.0.0.1:6013
0.0.0.0:445
127.0.0.1:6014
Active
10.211.55.5:22 = 10.211.55.2:61630 Established
10.211.55.5:22 = 10.211.55.2:50357 Established
10.211.55.5:22 = 10.211.55.2:50356 Established
10.211.55.5:22 = 10.211.55.2:57085 Established
10.211.55.5:22 = 10.211.55.2:60080 Established
```

### CODE REVIEW

**Comment by wfurt** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

File: src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Connections.cs [hunk 1]
```diff
@@ -80,21 +102,41 @@ internal static IPEndPoint[] ParseActiveTcpListenersFromFiles(string tcp4Connect
...
+                }
             }
 
             // TCP6 Connections
             for (int i = 1; i < v6connections.Length; i++) // Skip first line header.
             {
-                string line = v6connections[i];
-                IPEndPoint endPoint = ParseLocalConnectionInformation(line);
-                endPoints[index++] = endPoint;
+                TcpConnectionInformation ti = ParseTcpConnectionInformationFromLine(v4connections[i]);
```
Comment (DyachenkoA) [inline src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Connections.cs]: v4connections -> v6connections

File: src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Connections.cs [hunk 2]
```diff
@@ -80,21 +102,41 @@ internal static IPEndPoint[] ParseActiveTcpListenersFromFiles(string tcp4Connect
...
             for (int i = 1; i < v6connections.Length; i++) // Skip first line header.
             {
-                string line = v6connections[i];
-                IPEndPoint endPoint = ParseLocalConnectionInformation(line);
-                endPoints[index++] = endPoint;
+                TcpConnectionInformation ti = ParseTcpConnectionInformationFromLine(v6connections[i]);
+                if (ti.State == TcpState.Listen)
+                {
+                    endPoints[index] = ti.LocalEndPoint;
+                    index ++;
```
Comment (rmkerr) [inline src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Connections.cs]: nit: You have an extra space here and on line 114.

**Comment by wfurt** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

**Pull request merged**